mangaki/mangaki/views.py

```diff
@@ -293,12 +295,17 @@ def sort_mode(self):
         else:
             return sort
 
+    def is_dpp(self):
+        dpp = self.kwargs.get('dpp')
+        return dpp
+
     def get_queryset(self):
+        search_text = self.search()
         queryset = self.category.work_set.all()
         sort_mode = self.sort_mode()
-        search_text = self.search()
-
-        if sort_mode == 'top':
+        if self.is_dpp():
+            queryset = self.category.work_set.all().dpp(10)
```

Comment:
Il faut filtrer les `Work` qui ont déjà été notés par cet utilisateur ici ! La DPP me propose des oeuvres que j'ai déjà noté...

`self.category.work_set.exclude(coldstartrating__user=self.request.user).dpp(10)`